services/analyzer/app.py

```
@@ -49,109 +180,178 @@
 
+# ─────────────────────────────────────────────
+# Redis Helper
+# ─────────────────────────────────────────────
+async def redis_call(coro):
+    """Execute Redis coro with timeout + circuit breaker. Returns None on any failure."""
+    if not redis_client or not redis_cb.allow_request():
+        return None
+    try:
+        result = await asyncio.wait_for(coro, timeout=REDIS_TIMEOUT_S)
+        await redis_cb.record_success()
+        return result
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)